### ISSUE TITLE
Row.get(…) fails for array-typed values

### ISSUE THREAD

**Issue by mp911de** [r2dbc/r2dbc-postgresql#67]
Calling `Row.get(…)` or `Row.get(…, Object.class)` fails for array-typed columns with:

```
java.lang.IllegalArgumentException: Cannot decode value of type java.lang.Object
```

## Expectation

`Row.get(…)` should return the array value with the most appropriate type. For `INT[]` that should be `Integer[].class`.

## Steps to reproduce

SQL:

```sql
CREATE TABLE with_arrays (my_array INT[]);
INSERT INTO with_arrays (my_array) VALUES('{1,2,3}')
```

Code:

```java
Mono.from(connectionFactory.create())
    .flatMapMany(it -> {
        return it.createStatement("SELECT my_array FROM with_arrays").execute();
    }).flatMap(it -> it.map((row, rowMetadata) -> row.get("my_array")))
    .doOnNext(System.out::println)
    .as(StepVerifier::create)
    .expectNextCount(1)
    .verifyComplete();
```

**Comment by nebhale** [comment]
This may be because it’s [simply incomplete](
https://github.com/r2dbc/r2dbc-postgresql/pull/54).  Guess that one _does_ need to make it in for M7.

**Comment by mp911de** [comment]
There was some change between M6 and M7 as it has worked with M6.

**Comment by Squiry** [comment]
Well, I can fix this with a few lines, but the whole `Object.class` decoding (not only for arrays) thing stands on type erasure, I'm not sure how bad is that.

**Comment by nebhale** [comment]
Resolved by #69
